app/src/main/java/org/oppia/app/player/audio/AudioFragment.kt

```diff
@@ -52,4 +62,25 @@ class AudioFragment : InjectableFragment() {
     languageCodeList.add("hi-en")
     return languageCodeList
   }
+
+  private fun showCellularDataDialogFragment() {
+    cellularDataInterface = object : CellularDataInterface {
+      override fun cellularDataUserPreference(doNotShowAlert: Boolean, playAudio: Boolean) {
+        // true, true -> save preference and play audio
+        // true, false -> save preference and do not show AudioFragment
+        // false, true -> do not save preference and play audio
+        // false, false -> do nothing and do not show AudioFragment
+      }
+    }
+
+    val previousFragment = fragmentManager?.findFragmentByTag(TAG_CELLULAR_DATA_DIALOG)
+    if (previousFragment != null) {
+      fragmentManager?.beginTransaction()?.remove(previousFragment)?.commitNow()
+    }
+    val dialogFragment = CellularDataDialogFragment.newInstance(
+      cellularDataInterface
+    )
+    // Might need to replace show() with showNow().
```

Comment:
Let's use showNow(). I suspect it may not work because you're calling showCellularDataDialogFragment in onCreateView, and if I recall correctly you can't perform a fragment transaction during another fragment transaction.

We ought to not show a dialog immediately upon fragment creation. In fact, shouldn't the dialog only show when you try to play audio and you're on cellular?

Reply:
Regarding first point: Yes you are correct on fragment transaction point, it does it give error of fragment transaction already running.

Actually CellularDataDialogFragment should be visible even before we show AudioDialogFragment and based on user preference the audiobar will be visible.

I have put this here for testing purposes otherwise it might happen that when @jamesxu0 tries this code he may run into this error.

For now I will change this to `showNow` and add comment that for testing purpose only you can change this to `show`.

Reply:
Thanks for clarifying. After looking at the mocks again, it seems like we should this based on clicking the audio button in the player. I think that means we should be showing the cellular dialog at that point, and before we show the audio dialog. This also helps prevent the need to try and show a dialog during the lifecycle method of another fragment.

If the audio button checks for cellular connectivity and opens the cellular dialog, then it decides whether to proceed with showing the audio bar based on the user's input.

Do you agree?

Reply:
I agree on this completely, for now, I will add one dummy button on `ExplorationFragment`and on clicking on that it will show `CellularDataDialogFragment` from that if the response favours showing of `AudioFragment` then I will show that otherwise I will not show that.

Later on that dummy button in `ExplorationFragment` will be replaced with `item` in `toolbar`.

How does this sound?

Reply:
Yep sounds good--thanks!

Reply:
welcome.